cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1104,8 +1094,17 @@ protected Future<Void> rollingUpdate() {
         if (featureGates.useStrimziPodSetsEnabled())   {
             return maybeRollKafka(
                     podSetDiff.resource().getSpec().getPods().size(),
-                    pod -> ReconcilerUtils.reasonsToRestartPod(reconciliation, podSetDiff.resource(), pod, fsResizingRestartRequest, existingCertsChanged, clusterCa, clientsCa),
-                    listenerReconciliationResults.advertisedHostnames,
+                    pod -> ReconcilerUtils.reasonsToRestartPod(
+                            reconciliation,
+                            podSetDiff.resource(),
+                            pod,
+                            fsResizingRestartRequest,
+                            !Annotations.hasAnnotationWithValue(
```

Comment:
@scholzj this was missed in #7654 , this is to set the proper restart reason, that the cert has changed.